src/test/java/games/strategy/triplea/ui/TimespanDialogTest.java

```diff
@@ -0,0 +1,39 @@
+package games.strategy.triplea.ui;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.time.Instant;
+
+import org.junit.Test;
+
+import games.strategy.triplea.ui.TimespanDialog.TimeUnit;
+
+public class TimespanDialogTest {
+
+  @Test
+  public void testForeverReturnsNull() {
+    assertNull(TimeUnit.FOREVER.getInstant((int) (Math.random() * Integer.MAX_VALUE)));
+  }
+
+  @Test
+  public void testIntervalSizesBecomeBigger() {
+    final Instant minute = TimeUnit.MINUTES.getInstant(1);
+    final Instant hour = TimeUnit.HOURS.getInstant(1);
+    final Instant day = TimeUnit.DAYS.getInstant(1);
+    final Instant week = TimeUnit.WEEKS.getInstant(1);
+    final Instant month = TimeUnit.MONTHS.getInstant(1);
+    final Instant year = TimeUnit.YEARS.getInstant(1);
```

Comment:
Note that all these instants are potentially being created with a different start time since the call to `Instant#now()` is buried inside the `getInstant()` calls.  As we discussed before, tests shouldn't be dependent on wall time.  It might be better for `TimeUnit#getInstant()` to take the start time (in addition to the duration) and require the caller to pass it in.  That will push the `Instant#now()` calls into a single location within `TimespanDialog#prompt()`.  Something like this:

```diff
diff --git a/src/main/java/games/strategy/triplea/ui/TimespanDialog.java b/src/main/java/games/strategy/triplea/ui/TimespanDialog.java
index 92cfc30..3b6d97f 100644
--- a/src/main/java/games/strategy/triplea/ui/TimespanDialog.java
+++ b/src/main/java/games/strategy/triplea/ui/TimespanDialog.java
@@ -6,6 +6,7 @@ import java.time.LocalDateTime;
 import java.time.ZoneOffset;
 import java.time.temporal.ChronoUnit;
 import java.util.Date;
+import java.util.function.BiFunction;
 import java.util.function.Consumer;
 import java.util.function.Function;
 
@@ -30,18 +31,18 @@ public class TimespanDialog {
    */
   @VisibleForTesting
   enum TimeUnit {
-    MINUTES("Minutes", i -> Instant.now().plus(i, ChronoUnit.MINUTES)),
-    HOURS("Hours", i -> Instant.now().plus(i, ChronoUnit.HOURS)),
-    DAYS("Days", i -> Instant.now().plus(i, ChronoUnit.DAYS)),
-    WEEKS("Weeks", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.WEEKS).toInstant(ZoneOffset.UTC)),
-    MONTHS("Months", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.MONTHS).toInstant(ZoneOffset.UTC)),
-    YEARS("Years", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.YEARS).toInstant(ZoneOffset.UTC)),
-    FOREVER("Forever", i -> null);
+    MINUTES("Minutes", (start, i) -> start.plus(i, ChronoUnit.MINUTES)),
+    HOURS("Hours", (start, i) -> start.plus(i, ChronoUnit.HOURS)),
+    DAYS("Days", (start, i) -> start.plus(i, ChronoUnit.DAYS)),
+    WEEKS("Weeks", (start, i) -> LocalDateTime.ofInstant(start, ZoneOffset.UTC).plus(i, ChronoUnit.WEEKS).toInstant(ZoneOffset.UTC)),
+    MONTHS("Months", (start, i) -> LocalDateTime.ofInstant(start, ZoneOffset.UTC).plus(i, ChronoUnit.MONTHS).toInstant(ZoneOffset.UTC)),
+    YEARS("Years", (start, i) -> LocalDateTime.ofInstant(start, ZoneOffset.UTC).plus(i, ChronoUnit.YEARS).toInstant(ZoneOffset.UTC)),
+    FOREVER("Forever", (start, i) -> null);
 
     private final String name;
-    private final Function<Integer, Instant> function;
+    private final BiFunction<Instant, Integer, Instant> function;
 
-    private TimeUnit(final String name, final Function<Integer, Instant> supplier) {
+    private TimeUnit(final String name, final BiFunction<Instant, Integer, Instant> supplier) {
       this.name = name;
       this.function = supplier;
     }
@@ -52,8 +53,8 @@ public class TimespanDialog {
     }
 
     @VisibleForTesting
-    Instant getInstant(final Integer integer) {
-      return function.apply(integer);
+    Instant getInstant(final Instant start, final Integer integer) {
+      return function.apply(start, integer);
     }
   }
 
@@ -79,7 +80,7 @@ public class TimespanDialog {
             .build())
         .build(), title, JOptionPane.OK_CANCEL_OPTION);
     if (returnType == JOptionPane.OK_OPTION) {
-      final Instant instant = ((TimeUnit) comboBox.getSelectedItem()).getInstant((Integer) spinner.getValue());
+      final Instant instant = ((TimeUnit) comboBox.getSelectedItem()).getInstant(Instant.now(), (Integer) spinner.getValue());
       action.accept(instant == null ? null : Date.from(instant));
     }
   }
```